Extensions/IISWebAppDeploy/Src/Tasks/SqlDacpacDeploy/DeployToSqlServer.ps1

```diff
@@ -0,0 +1,131 @@
+﻿Import-Module $env:CURRENT_TASK_ROOTDIR\DeploymentSDK\InvokeRemoteDeployment.ps1
+
+function EscapeSpecialChars
+{
+    param(
+        [string]$str
+    )
+
+    return $str.Replace('`', '``').Replace('"', '`"').Replace('$', '`$')
+}
+
+function IsPublishProfileEmpty
+{
+    param(
+        [string]$publishProfile
+    )
+
+    $publishProfile = $publishProfile.Trim('\', ' ').Trim()
+
+    return ([string]::IsNullOrWhitespace($PublishProfile) -or
+    $PublishProfile -eq $env:SYSTEM_DEFAULTWORKINGDIRECTORY )
+}
+
+function TrimInputs([ref]$adminUserName, [ref]$sqlUsername, [ref]$dacpacFile, [ref]$publishProfile)
+{
+    Write-Verbose "Triming inputs for excess spaces, double quotes"
+
+    $adminUserName.Value = $adminUserName.Value.Trim()
+    $sqlUsername.Value = $sqlUsername.Value.Trim()    
+
+    $dacpacFile.Value = $dacpacFile.Value.Trim('"', ' ')
+    $publishProfile.Value = $publishProfile.Value.Trim('"', ' ')
+}
+
+function RunRemoteDeployment
+{
+    param(
+        [string]$machinesList,
+        [string]$scriptToRun,
+        [string]$adminUserName,
+        [string]$adminPassword,
+        [string]$winrmProtocol,
+        [string]$testCertificate,
+        [string]$deployInParallel
+    )
+
+    Write-Host "Starting deployment of Sql Dacpac File : $dacpacFile"
+
+    $errorMessage = Invoke-RemoteDeployment -machinesList $machinesList -scriptToRun $scriptToRun -adminUserName $adminUserName -adminPassword $adminPassword -protocol $winrmProtocol -testCertificate $testCertificate -deployInParallel $deployInParallel
+
+    if(-not [string]::IsNullOrEmpty($errorMessage))
+    {
+        $helpMessage = "For more info please refer to http://aka.ms/sqlserverdacpackreadme)"
+        Write-Error "$errorMessage`n$helpMessage"
+        return
+    }
+
+    Write-Host "Successfully deployed Sql Dacpac File : $dacpacFile"
+}
+
+function GetScriptToRun
+{
+    param (
+        [string]$dacpacFile,
+        [string]$targetMethod,
+        [string]$serverName,
+        [string]$databaseName,
+        [string]$sqlUserName,
+        [string]$sqlPassword,
+        [string]$connectionString,
+        [string]$publishProfile,
+        [string]$additionalArguments
+    )
+
+    $sqlPackageScript = Get-Content .\SqlPackageOnTargetMachines.ps1 | Out-String
+
+    $connectionString = EscapeSpecialChars -str $connectionString
+    $sqlPassword = EscapeSpecialChars -str $sqlPassword
+    $additionalArguments = EscapeSpecialChars -str $additionalArguments
```

Comment:
Inconsistency with TrimInputs.
TrimInputs was called once on all the required params. Here we are calling multiple times